server/schemas/resolvers.js

```diff
@@ -70,6 +70,13 @@ const resolvers = {
       const teacher = await Teacher.findOne({ email });
       const student = await Student.findOne({ email });
 
+      if (!email) {
+        throw new Error("Please enter your email");
+      }
+      if (!password) {
+        throw new Error("Please enter your password");
+      }
+
       const user = teacher || student;
```

Comment:
I've defined the errors to be show in the UI here in the resolver. Again, I'm not sure if this is missing the point but it's the closest I could get to it at the moment.